spec-2025x.md.html

```diff
@@ -0,0 +1,1498 @@
+
+Thin-film interference manifests as an iridescent effect resulting from the interference between light waves that scatter in a wavelength-dependent manner when interacting with films of thickness close to visible wavelengths. This interference occurs at both the upper and lower interfaces of the thin layer, influencing whether reflected light is suppressed or enhanced.
+
+The appearance of the thin-film effect is controlled by the film's thickness $t_i$ and its IoR $\eta_i$.
+
+| Name     | Type  | Default | Range   | Description                                  |
+|----------|-------|---------|---------|----------------------------------------------|
+| $i$      | float | 0.0     | [0,1]   | Weight of the thin-film layer.               |
+| $d_i$    | float | 0.4     | [0,1]   | Thickness of the thin-film layer in microns. |
+| $\eta_i$ | float | 1.3     | [1,2.5] | Index of refraction of the thin-film layer.  |
+[Table [thin-film-parameters]: List of thin-film parameters.]
+
+![Figure [thin-film-parameters]: Metal spheres with thin-film coating.](img/params_thin_film_2025x.png)
+
+We model the thin-film effect by introducing a thin layer on top of our core material, as proposed by [#BB17]. This layer approximates iridescence by a physically plausible simulation, leveraging the principles of wave optics. Evaluating it effectively replaces our Fresnel term $F(cos\Theta, f_0, f_{90})$ (Section [Fresnel]) with a more intricate term $R(cos\Theta, \eta, d_i, \eta_i)$, the thin-film reflectance. The new term comprehensively considers all inter-reflections inside the thin-film layer, encompassing constructive and destructive interference effects.
+
+Based on the thin-film layer weight $i$, we blend the original Fresnel with the thin-film reflectance for the specular core components
+$$
+OpaqueCore(...) = \mathbf{M}_r (1-i)F(...) + \mathbf{M}_r i R(...)...
+$$
+
+Likewise for transmission
+$$
+TransparentCore(...) = \mathbf{M}_t (1-i)(1-sF(...)) + \mathbf{M}_r i (1-R(...))...
+$$
+
+!!!
+    Please note how the specular weight doesn't affect the thin-film reflectance. It's obvious in the transmission equation above, but also true for the reflection as it uses the actual base IoR instead of the $f_0$ term.
+
+The diffuse BSDF is defined as a blend of the energy correct diffuse BSDF $B_c$ as given in eqn. [b_c] and a new term $B_{c,i}$
+
+\begin{equation}
+\begin{aligned}
+B(\mathbf{v}, \mathbf{l};...) = (1-i)B_c(\mathbf{v}, \mathbf{l}; ...) + i\cdot B_{c,i}(\mathbf{v}, \mathbf{l}; ...)
+\end{aligned}
+\end{equation}
+
+with
+
+\begin{equation}
+\begin{aligned}
+B_{c,i}(\mathbf{v}, \mathbf{l};...) = B_u(\mathbf{v}, \mathbf{l}; ...)( 1 - max(1-E_i(\mathbf{v}\cdot\mathbf{n};...), 1-E_i(\mathbf{l}\cdot\mathbf{n};...))).
```

Comment:
Is it possible to show analytically why this works? If so, would be nice to add the derivation to the Appendix 6.2. At least show in a few plots/tables that the "real" E_i, computed numerically, is greater than the approximative E_i used here (side-note: might want to call it $\hat{E}_i$ to differentiate the approximation from the actual $E_i$).